source-zoho-books contribution from bishalbera

bishalbera opened the pull request:
## What

This PR adds a new connector for source Zoho Books (source-zoho-books).

Contributor-provided connector description:

The Zoho Books connector  enables seamless integration of financial data, automating the flow of invoices, payments, expenses, and bank transactions into your data systems. With this connector, businesses can streamline bookkeeping processes, ensuring accurate financial reporting and real-time insights.
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `region` | `string` | Region. The region code for the Zoho Books API, such as &#39;com&#39;, &#39;eu&#39;, &#39;in&#39;, etc. |  |
| `client_id` | `string` | OAuth Client ID.  |  |
| `client_secret` | `string` | OAuth Client Secret.  |  |
| `refresh_token` | `string` | OAuth Refresh Token.  |  |
| `start_date` | `string` | Start date.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| users | user_id | DefaultPaginator | ✅ |  ❌  |
| banking | account_id | DefaultPaginator | ✅ |  ❌  |
| organizations | organization_id | DefaultPaginator | ✅ |  ❌  |
| contacts | contact_id | DefaultPaginator | ✅ |  ❌  |
| bills | bill_id | DefaultPaginator | ✅ |  ✅  |
| estimates | estimate_id | DefaultPaginator | ✅ |  ✅  |
| items | item_id | DefaultPaginator | ✅ |  ❌  |
| invoices | invoice_id | DefaultPaginator | ✅ |  ✅  |
| expenses | expense_id | DefaultPaginator | ✅ |  ✅  |
| creditnotes | creditnote_id | DefaultPaginator | ✅ |  ✅  |
| customerpayments | payment_id | DefaultPaginator | ✅ |  ✅  |
| purchaseorders | purchaseorder_id | DefaultPaginator | ✅ |  ✅  |
| salesorders | salesorder_id | DefaultPaginator | ✅ |  ✅  |
| journals | journal_id | DefaultPaginator | ✅ |  ✅  |
| taxes | tax_id | DefaultPaginator | ✅ |  ❌  |
| transactions |  | DefaultPaginator | ✅ |  ✅  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

<!-- DO NOT REMOVE: connector-builder-new-connector-contribution -->

marcosmarxm reviewed (changes requested):
Make `region` a enum type where user can select from a predefined list.

![image](https://github.com/user-attachments/assets/5df6052d-39cf-48b3-824e-f838724739fe)
You need to inject the `page_size` parameter to all streams.

Some endpoints like salesorders support incremental, do you want to implement them?

bishalbera commented:
> Make `region` a enum type where user can select from a predefined list.
> 
> ![image](https://private-user-images.githubusercontent.com/5154322/377955887-5df6052d-39cf-48b3-824e-f838724739fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyOTg2NjgsIm5iZiI6MTcyOTI5ODM2OCwicGF0aCI6Ii81MTU0MzIyLzM3Nzk1NTg4Ny01ZGY2MDUyZC0zOWNmLTQ4YjMtODI0ZS1mODM4NzI0NzM5ZmUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTlUMDAzOTI4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzI1OWJlMTAyZjRlZTVkMWZjOGM3YzBmZTUxNzVlN2NiNGU4MDA1NmZhMzA4YmQ4ZTdjZmM1NGQ5YzRjMDNkMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.GF1hbt1d0u7QCqyso-JNiJuRXpQffwjZUzUEYSH0EA4) You need to inject the `page_size` parameter to all streams.
> 
> Some endpoints like salesorders support incremental, do you want to implement them?

fixed pagination and added incremental to supported streams . Please verify it so that I can apply it to the zoho inventory issue also as they are identical. @marcosmarxm

marcosmarxm commented:
Hello 👋 Due to the numerous contributions happening, I decided to create this [view](https://github.com/orgs/airbytehq/projects/104/views/3) to help you know when I plan to review your contribution. If your connector has some limited-time credential feel free to reach me out.
